components/button/button-subtle.js

```diff
@@ -213,6 +220,19 @@ class ButtonSubtle extends ButtonMixin(RtlMixin(LitElement)) {
 		`;
 	}
 
+	willUpdate(changedProperties) {
+		if (changedProperties.has('_hasCustomIcon') || changedProperties.has('icon')) {
+			this._hasIcon = this._hasCustomIcon || this.icon;
+		}
+	}
+
+	_handleIconSlotChange(e) {
+		const icon = e && e.target && e.target.assignedNodes({ flatten: true }).filter((node) => {
+			return node.nodeType === Node.ELEMENT_NODE && node.tagName === 'D2L-ICON-CUSTOM';
+		});
+		this._hasCustomIcon = icon.length === 1;
```

Comment:
I _think_ everything can be simplified a bit to just require a single `_hasCustomIcon` state property? If you apply a CSS class in `render()` directly to the button as Dave suggested when `_hasCustomIcon || this.icon` is true, then I don't think you need the `willUpdate` code or the `_hasIcon` property. Could be missing something though.

Reply:
Yup I think that's right